src/mailmap.c

```diff
@@ -0,0 +1,381 @@
+/*
+ * Copyright (C) the libgit2 contributors. All rights reserved.
+ *
+ * This file is part of libgit2, distributed under the GNU GPL v2 with
+ * a Linking Exception. For full terms see the included COPYING file.
+ */
+
+#include "git2/mailmap.h"
+
+#include "blob.h"
+#include "commit.h"
+#include "git2/common.h"
+#include "git2/repository.h"
+#include "git2/revparse.h"
+#include "git2/sys/commit.h"
+
+/**
+ * Helper type and methods for the mailmap parser
+ */
```

Comment:
No need for Javadoc-style comments in implementations.

Furthermore, we do have a rather new `git_parse` interface now which should preferably be used to parse data. We want to avoid duplicating common parser functions and structures as much as possible. Have you considered using that interface?

Reply:
I wasn't aware that the `git_parse` interface existed :-). I grepped around the codebase looking at existing parsers & saw that they seemed to be implemented in an ad-hoc way, so did the same here.

I'll look into using that interface instead.